libbpf-tools/trace_helpers.c

```diff
@@ -1247,3 +1247,42 @@ bool probe_ringbuf()
 	close(map_fd);
 	return true;
 }
+
+int split_convert(char *s, const char* delim, void *elems, size_t elems_size,
+		  size_t elem_size, convert_fn_t convert)
+{
+	char *token;
+	int ret;
+	char *pos = (char *)elems;
+
+	if (!s || !delim || !elems)
+		return -EINVAL;
+
+	errno = 0;
+	token = strtok(s, delim);
+	while (token) {
+		if (pos >= (char*)elems + elems_size)
+			return -ENOBUFS;
+
+		ret = convert(token, pos);
+		if (ret)
+			return -ret;
+
+		pos += elem_size;
+		token = strtok(NULL, delim);
+	}
+
+	return 0;
+}
+
+int str_to_int(const char *src, void *dest)
+{
+	*(int*)dest = strtol(src, NULL, 10);
+	return errno;
```

Comment:
Since str_to_int() and str_to_long() are in trace_helpers.c, we should add 'error = 0;' before strtol(). It is possible that errno is not 0 before strtol() and if strtol() works perfectly, errno won't change and the return value will be non-zero.